app/controllers/application_controller.rb

```diff
@@ -5,6 +5,146 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+		set :session_secret, "password_security"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/failure' do
+		erb :failure
+	end
+
+  get '/login' do
+    if logged_in?
+      redirect '/tweets'
+    else
+	    erb :'/users/login'
+    end
+	end
+
+	get '/logout' do
+		session.clear
+		redirect '/login'
+	end
+
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'/users/signup'
+    end
+	end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'/tweets/create_tweet'
+    else
+	    redirect '/login'
+    end
+	end
+
+  get '/tweets/:id/delete' do
+    if logged_in?
+      tweet = Tweet.find(params[:id])
+      tweet.delete
+    else
+	    redirect '/login'
+    end
+	end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` (and not just that any user is logged in) before allowing them to edit?